### ISSUE TITLE
[Feature]: Enable developers to remove the appsmith watermark

### ISSUE THREAD

**Issue by dilippitchika** [appsmithorg/appsmith#14143]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

**Job stories**
When I have paid for appsmith, I want to disable appsmith watermark so that my team mates trust the interface i built on appsmith more

When I have paid for appsmith as a consultant, I want to disable appsmith watermark so that my clients don't know which tools I use to build their apps.

### Why should this be worked on?

Users who have paid for appsmith should have the option to disable appsmith watermark if they are not interested in letting others know about the tool they are using to build apps.

**Comment by zerofill** [comment]
Curious to what the reasoning is for the logo in there anyway. I think it is to promote the appsmith platform. But advertising a platform to build apps, inside of an app, has never actually worked. You can remove it inside of the docker container, but that is not something users would really want to do. When you embed a appsmith page inside of another app, it just looks terrible.

So I am for the logo removal, at least removed when the embed tag is used.

**Comment by dilippitchika** [comment]
@zerofill thanks for the insight, could you expand on this
> When you embed a appsmith page inside of another app, it just looks terrible.
Could you share your screenshot for this? I would like to know more about it.

**Comment by zerofill** [comment]
Well I can't really share a screenshot of this app. Without basically editing everything out of it lol. But, you have a sidebar menu. Then you have a header... My nav on the left and the top header display some info but they are blended in so it all looks like 1 solid page as far as color etc... Then you have a footer. When you embed a page, it looks like some "Built on" white box floating over on the right. Just looks bad, so I end up removing it.

Not to mention the application isn't built on appsmith. I am just using it to add some things on certain pages. it is like 3% of the project. It has a massive php built api, and the site is built in php. The API all hand written by me, the app other than the CodeIgniter framework (which I am not a huge fan of) but it gets out of my way... that is all written line by line too.

At one point I was going to do the whole application in appsmith. But, I couldn't have the header, etc... all having appsmith all over every page. Because it isn't an internal application. This is a SaaS application. That, and I am a php guy not a javascript guy. So it was easier for me to do most of it in php with ajax. Eventually probably converting all the ajax to vue or something in the future.

**Comment by dilippitchika** [comment]
@zerofill thanks for the extra info, i think I understand this better now. 

I have a few more questions mostly on your usecase and why you are using appsmith given that you are uncomfortable with JS. Will you be ok to have a short call with me and also show your app? If yes could you please block a slot which works for you - https://calendly.com/dilip_pitchika/30min

**Comment by zerofill** [comment]
Not so much uncomfortable with javascript, as in I don't really use it to develop full applications. Mainly just for updating divs from ajax calls to class functions in controllers. The syntax, etc... in appsmith can cause me issues haha. The main reason for using it at the moment, was to speed the release process up.

I am the only developer on it. The project is an A.I. related project. There are 2 companies involved in it. But, I am the sole developer, because what I have had to go through learning, and testing, and trial and error, is just impossible to catch anyone up on. Without slowing the process down. We had someone from Lockheed Martin working with us in his free time. But, it just didn't work out. There was too much he wanted to understand about the app, and it was too much of a time sync to catch him up on it.

I honestly wouldn't have a problem sticking something like a message in the footer of the application that said some components you see on some pages were created using the great tool appsmith, or something like that with a link.

Unfortunately the application itself, I can't show to anyone yet.

I would have no problem showing you it when it is released.

I had a lot done in a previous version of Appsmith, and then don't know what the hell I did with the export haha.
I basically when back to PHP/CSS/AJAX to build the app because it was nearly impossible to get the look I wanted at the time. Maybe it is possible now. So as of yesterday I spun up appsmith again. Because There were certain things I just don't want to spend the time on right now making controllers, routes, ajax calls, etc... for.

So I am pretty sure I can complete those in appsmith and embed them for now. There is some info I am storing when they login to the PHP app for the moment, that is base64. Using it kinda like a token, that I want to pass to the embedded pages in appsmith. So I have to figure that out. My API does jwt with firebase/php-jwt. However, I obviously don't want them to re-authenticate, when they are only going to another page in the application. I probably could have did a handoff, if I was doing JWT authentication in the web application. But, it is using sessions, and an API key for certain actions.

Anyway... Like I was saying, I can't show the app until it is released. I would be happy to do that after it is released, if you wanted to record something to further promote appsmith, etc... To use for YouTube, or whatever.

If you have more questions, we could also do that in email or something as well.

Going to stop now, because I am starting to write an ebook by the looks of it haha.

**Comment by vivekverma2312** [comment]
I will test this once this [PR](https://github.com/appsmithorg/appsmith-ee/pull/390) is merged in release.

**Comment by okaeiz** [comment]
so I cannot remove the watermark, can I?

**Comment by dilippitchika** [comment]
@okaeiz this feature is available for our paid users. Have you already bought appsmith?

**Comment by Ghafrigm** [comment]
I think he is talking specifically about the self hosted option. If for example I purchase the paid option how can I remove the logo if I want to have it hosted as a self host?

**Comment by dilippitchika** [comment]
@Ghafrigm that feature is available in admin settings for self hosted paid users. We are fixing some bugs there.

**Comment by Ghafrigm** [comment]
> @Ghafrigm that feature is available in admin settings for self hosted paid users. We are fixing some bugs there.

It doesn't make sense to pay to have a white label. Even though competitors offer that option from the start with self hosting already. Its one of the reasons I switched actually, but will see if I can do it from the source anyways.

**Comment by dilippitchika** [comment]
Hey @Ghafrigm thanks for your feedback, could you expand why you feel this way? Also which competitor are you referring to here?

**Comment by bubelarek** [comment]
Hi,
Some JS tweak to do it:
1/ Add Iframe at bottom of app
2/ Put this code in srcDoc
`<html>
<body>
<script>
//alert()
window.parent.document.querySelectorAll('a[href="https://appsmith.com"').forEach(e => e.remove());
//window.parent.document.querySelectorAll('#header-root').forEach(e => e.remove());
window.parent.document.querySelectorAll('div.t--widget-iframewidget').forEach(e => e.remove());
</script>
</body>`
3/ this code will remove logo and iframe itself when app is running.
4/ it will also remove iframe from view in edit mode, but you have it in editor on the left.

**Comment by wemod123** [comment]
@bubelarek @appsmith-org
Will this have license issue with appsmith org? 
I mean this feature is said paid feature while you remove it without agreements from appsmith org,  is this OK to go?

**Comment by sergioponguta** [comment]
> 

Seems like it doesn't work anymore

**Comment by bubelarek** [comment]
> @bubelarek @appsmith-org Will this have license issue with appsmith org? I mean this feature is said paid feature while you remove it without agreements from appsmith org, is this OK to go?

Appsmith is mostly Apache licence so, it is ok to even replace logo with you own and start selling Appsmith  clone. Some opensource  projekt move to AGPL 3.0 becouse at least you cannot modify and sell without sharing. At least  it is my understanding

**Comment by passos** [comment]
I wonder why it's hard for developers to take off logos from apps they host themselves, even when they have all the code. I remember seeing a few years ago that some companies worked really hard to lock their code to stop developers from doing this. It's funny because this goes against the idea of "open-source," which is supposed to let people change and use the code freely.
